pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -56,12 +56,16 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
 	outputPodsList.Items = []v1.Pod{}
 
 	for _, pod := range inputPodsList.Items {
+		if _, ok := pod.Annotations["sidecar.istio.io/status"]; !ok {
+			continue
+		}
+		if !isPodReady(pod){
+			continue
+		}
```

Comment:
You could somehow merge these two conditions in one sentence to avoid multiple branching with the same result (both end with `continue`).